src/http/proxy_wasm/ngx_http_proxy_wasm_dispatch.c

```diff
@@ -933,6 +915,15 @@ ngx_http_proxy_wasm_dispatch_resume_handler(ngx_wasm_socket_tcp_t *sock)
         ngx_proxy_wasm_ctx_set_next_action(pwexec->parent,
                                            NGX_PROXY_WASM_ACTION_PAUSE);
 
+    } else if (ngx_proxy_wasm_foreign_calls_total(pwexec)) {
+        ngx_log_debug0(NGX_LOG_DEBUG_WASM, pwexec->log, 0,
+                       "proxy_wasm foreign function callbacks pending...");
+
+        rc = NGX_AGAIN;
+        ngx_wasm_yield(&rctx->env);
+        ngx_proxy_wasm_ctx_set_next_action(pwexec->parent,
+                                           NGX_PROXY_WASM_ACTION_PAUSE);
```

Comment:
When I further modified this file to support yieldable foreign calls from `on_http_call_response` I became more apparent to me the duplication concern you've shared a few times.

Yup, combining the two queues into a single one will make the code simpler and more concise, thanks for suggesting!

Reply:
It may be tricky to store both foreign and http dispatches in the same queue. Perhaps a single type with a `union` member could combine both HTTP calls and foreign calls, and that would be what is stored in the single dispatch_queue?